lib/utils.js

```diff
@@ -131,30 +131,60 @@ exports.isDir = function (path) {
 };
 
 
+// Given a path, determine if the path is a regular file
+exports.isFile = function (path) {
+
+    try {
+        var stat = Fs.statSync(path);
+        return stat.isFile();
+    }
+    catch (e) {
+        return false;
+    }
+};
+
+
 // Given a starting directory, find the root of a git repository.
-// In this case, the root is defined as the first directory that contains
-// a directory named ".git"
 //
-// Returns a string if found, otherwise undefined
+// Returns undefined if not found, otherwise an object containing:
+//   root: path to the root of the Git repo
+//     - this is the first directory (traversing upward) that contains a ".git"
+//       directory (most cases) or a ".git" file (submodule inside another Git repo)
+//   gitDir: path to the ".git"-ish directory of the Git repo
+//     - in most cases, this is the ".git" directory inside the Git repo root
+//     - in the submodule case, this is the directory pointed to by the ".git" file
 exports.findGitRoot = function (start) {
 
     start = start || Path.dirname(internals.findParent(module).filename);
     var root;
+    var gitDir = Path.join(start, '.git');
 
-    if (exports.isDir(Path.join(start, '.git'))) {
+    if (exports.isDir(gitDir)) {
+        root = start;
+    }
+    else if (exports.isFile(gitDir)) {
+        // This is a git submodule inside another git repo!
+        // Resolve the path to its ".git"-ish directory.
         root = start;
+        gitFile = Fs.readFileSync(gitDir, { encoding: 'utf8' });
+        matches = gitFile.split('\n')[0].match(/^gitdir: (.+)$/);
```

Comment:
likewise need a `var` here. also do we really need to split on newlines? why not just use the regex?